### DISCUSSION TITLE
reactive ten, effect must watch ten all together?

### DISCUSSION THREAD

**Discussion by ericxin1982** [alpinejs/alpine#4130] [1. Help]
![image](https://github.com/alpinejs/alpine/assets/9047436/ddbefcfb-43b3-49b4-a31b-122f4bd9fa54)
Hi,

From this example, one reactive to one effect, I can get it.
But if I have ten reactive, and I only care specific one in one of the x-data scope to keep updating the latest value, how to do?
Or the effect is like global watcher for all defined reactive?

Thanks
Eric Xin

**Comment by ericxin1982** [comment]
![image](https://github.com/alpinejs/alpine/assets/9047436/1a1b1f0b-d6c9-4407-ac3c-bb7eb4b4b6d7)
![image](https://github.com/alpinejs/alpine/assets/9047436/334df067-eb84-4969-a83f-d23e04bfd067)
![image](https://github.com/alpinejs/alpine/assets/9047436/0485b982-88a4-46ee-a8dd-0e6673115902)
![image](https://github.com/alpinejs/alpine/assets/9047436/6a14255e-7a74-4fc5-839b-291eed929424)

Like this, if effect must watch all, how I can watch each individual respectively? I do not want them to get influenced across. 

Thanks
Eric Xin

**Comment by SimoTod** [comment]
You can add as many separate Alpine.effect callback as you want in your script (unless I misunderstood the question).

**Reply by ericxin1982** [reply]
Yes, you are right.
![image](https://github.com/alpinejs/alpine/assets/9047436/e49edcb9-8dd2-4c44-917d-5139652bfe09)
I just thought it would overwrite previous one.
That is what I want.

**Comment by ekwoka** [comment]
Effects can respond to unlimited reactive values, and reactive values can be watched by unlimited effects.

Normally these apis of `Alpine.reactive` and `Alpine.effect` (especially reactive) are not part of "normal" simple use of Alpine, but more advanced usage or plugin design.

If you're still new to Alpine, it would smarter to just understand they are there, as opposed to trying to build something around them.

You can almost see it as Alpine just being a kind framework and exposing access to some basic internals.

**Reply by ericxin1982** [reply]
reactive is an object has many properties,  but effect just want to be respective, I did not try each single piece, just ask senior person first.
